File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Java/src/main/java/gameCatan/factories/MapFactory.java
package gameCatan.factories;

import gameCatan.models.Box;
import gameCatan.models.Map;

import static gameCatan.utils.funciones.checkTwiceResources;

public class MapFactory {
    public static Map create() {
        int sizeRow = 3;
        int sizeCol = 4;
        Box[][] boxes = new Box[sizeRow][sizeCol];
        do {
            for (int i = 0; i < boxes.length; i++) {
                for (int j = 0; j < boxes[i].length; j++) {
                    boxes[i][j] = BoxFactory.create();
                }
            }
        } while (!checkTwiceResources(boxes));
        return new Map(sizeRow, sizeCol, boxes);
    }
}

File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Kotlin/gameCatan/factories/MapFactory.kt
package gameCatan.factories

import gameCatan.models.Box
import gameCatan.models.Map
import gameCatan.utils.funciones.checkTwiceResources

object MapFactory {
    @JvmStatic
    fun create(): Map {
        val sizeRow = 3
        val sizeCol = 4
        val mapOfBoxes = Array(sizeRow) { arrayOfNulls<Box>(sizeCol) }
        do {
            for (i in mapOfBoxes.indices) {
                for (j in mapOfBoxes[i].indices) {
                    mapOfBoxes[i][j] = BoxFactory.create()
                }
            }
        } while (!checkTwiceResources(mapOfBoxes))

        return Map(sizeRow, sizeCol, mapOfBoxes)
    }
}